### PR TITLE
Soen19 login navbar

### PR DESCRIPTION
**Pull request by KelvinTrang**
# Description
We linked the backend and frontend for the login and signup page. New users can be created if all validations are fine (email, password, postal code, and missing fields).  

Fixes [SOEN-48](https://soen390team8.atlassian.net/browse/SOEN-48?atlOrigin=eyJpIjoiZTdlNmE1YmE3NmNlNDI2OGFlNTdkZWY4OGM5MzMzNDIiLCJwIjoiaiJ9)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Test manually. The testing framework has not been implemented.
- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: backend/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,7156 @@
+{
```
Comment (jfv-2000) [inline backend/package-lock.json]: This file needs to be untracked so that it isn't pushed into main

File: frontend/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,8843 @@
+{
```
Comment (jfv-2000) [inline frontend/package-lock.json]: This file must be untracked as well so that it isn't pushed into the main branch

**Review by jfv-2000 — CHANGES_REQUESTED** [review]
Very good work BingBong 👍

**Review by jfv-2000 — APPROVED** [review]

**Review by amqkv — APPROVED** [review]
Tested & works well.
Very good work BingBong 👍

File: frontend/components/inputs/DateInput.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { Text } from "@chakra-ui/react";
+import { InputProps } from "@frontend/components/inputs/types/input";
+
```
Comment (stevenvuong00) [inline frontend/components/inputs/DateInput.tsx:3]: not enough comments to know exactly what this is

File: frontend/package.json [hunk 4]
```diff
@@ -12,6 +12,8 @@
     "@chakra-ui/react": "^1.7.4",
     "@emotion/react": "^11.7.1",
     "@emotion/styled": "^11.6.0",
+    "date-fns": "^2.28.0",
```
Comment (stevenvuong00) [inline frontend/package.json]: are these packages really used? they should be removed.

File: frontend/pages/auth/signin.tsx [hunk 5]
```diff
@@ -10,10 +12,31 @@ import {
...
+} from "@frontend/utils/validation";
+import { signUp } from "@frontend/functions/signUp";
+import {
+  registerGeneralErrorPopup,
+  registerSuccessPopup,
+  signinErrorPopup,
+  registerEmailErrorPopup,
+} from "@frontend/utils/popups";
 
+const today = new Date().toISOString().slice(0, 10);
```
Comment (stevenvuong00) [inline frontend/pages/auth/signin.tsx]: i don't think this is being used in the file

File: frontend/pages/auth/signin.tsx [hunk 6]
```diff
@@ -23,20 +46,104 @@ export async function getServerSideProps(context: any) {
...
+  const callPopup = (props: UseToastOptions) => !toast.isActive("popup") && toast({ ...props, id: "popup" });
+
+  const [emailError, setEmailError] = useState(false);
+  const [passwordError, setPasswordError] = useState(false);
+  const [confirmPasswordError, setConfirmPasswordError] = useState(false);
+  // const [phoneError, setPhoneError] = useState(false);
+  const [postalError, setPostalError] = useState(false);
+  const [tabIndex, setTabIndex] = useState(0);
+
+  async function handleSignIn(event: any) {
```
Comment (stevenvuong00) [inline frontend/pages/auth/signin.tsx:59]: i don't like this any. could be more specific

File: frontend/pages/auth/signin.tsx [hunk 7]
```diff
@@ -45,43 +152,95 @@ export default function SignIn({ csrfToken }: any) {
...
+                        name="Confirm Password"
+                        placeholder="Enter Password Again"
+                        label="Confirm Password"
+                        error={confirmPasswordError}
+                      />
+                      <DropdownInput
+                        placeholder="Select Account Role"
+                        name="Account Role"
+                        label="Account Role"
+                        options={["Doctor", "Health Official", "Immigration Officer", "Patient"]}
```
Comment (stevenvuong00) [inline frontend/pages/auth/signin.tsx]: maybe this should be moved to a constant?

File: frontend/pages/auth/signin.tsx [hunk 8]
```diff
@@ -23,20 +46,104 @@ export async function getServerSideProps(context: any) {
...
+      setEmailError(true);
+      error = true;
+    } else setEmailError(false);
+
+    if (!validPostalCode(registerValues.postalCode)) {
+      setPostalError(true);
+      error = true;
+    } else setPostalError(false);
+
+    // if someone wants to do phone validation need to find regex that accepts 5141234567
```
Comment (silentswordxv) [inline frontend/pages/auth/signin.tsx]: use /^\d{3}[-. ]?\d{3}[-. ]?\d{4}$/

File: frontend/utils/validation.ts [hunk 9]
```diff
@@ -0,0 +1,22 @@
+// export function validPhoneNumber(phoneNumber: string) {
+//   return /^((\+1)?[\s-]?)?\(?[2-9]\d\d\)?[\s-]?[2-9]\d\d[\s-]?\d\d\d\d/.test(phoneNumber);
+// }
+
+export function validPassword(password: string) {
+  return /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/.test(password);
```
Comment (stevenvuong00) [inline frontend/utils/validation.ts:6]: i'm just going to trust you on this that it works.

**Review by silentswordxv — CHANGES_REQUESTED** [review]
Implement my regex :)

**Review by stevenvuong00 — COMMENTED** [review]
looks good, little changes here and there but good work!

**Pull request merged**